Exception stack trace truncated — cause chain not fully reported

## Summary

*Generated by Claude*

When an exception propagates out of a cell, JJava truncates the cause chain. Only the top-level wrapper exception and one level of `Caused by` (a `jdk.jshell.EvalException`) are shown. Any deeper causes — including the true root cause — are silently dropped, making it very hard to diagnose errors.

## Steps to Reproduce

Run any code that throws an exception with a multi-level cause chain. A typical scenario is calling a method that catches a low-level exception (e.g. a `NullPointerException` inside a third-party library), wraps it in a domain exception, and throws that — producing a chain of two or more `Caused by` levels.

## Actual Behavior

JJava prints only two levels of the exception chain:

```
java.lang.RuntimeException: com.example.MyServiceException, Operation failed
|       at org.dflib.jjava.kernel.execution.CodeEvaluator.evalSingle(CodeEvaluator.java:145)
|       at org.dflib.jjava.kernel.execution.CodeEvaluator.eval(CodeEvaluator.java:79)
|       at org.dflib.jjava.kernel.JavaKernel.doEval(JavaKernel.java:230)
|       ...
|   Caused by: jdk.jshell.EvalException: Operation failed
|       at com.example.client.MyClient.performAction(MyClient.java:116)
|       at com.example.client.MyClient.execute(MyClient.java:46)
|       at .(<Anonymous>#61:1)
```

The exception chain ends here. The two deeper `Caused by` frames — the intermediate library error and the root-cause `NullPointerException` — are missing entirely.

## Expected Behavior

The full exception chain should be reported, matching what the JVM itself produces:

```
com.example.MyServiceException: Operation failed
    at com.example.client.MyClient.performAction(MyClient.java:116)
    at com.example.client.MyClient.execute(MyClient.java:46)
    ...
Caused by: Cannot invoke "String.length()" because "this.input" is null
    at com.thirdparty.lib.SomeClient.doRequest(SomeClient.java:161)
    at com.example.client.MyClient.performAction(MyClient.java:114)
    ... 9 more
Caused by: java.lang.NullPointerException: Cannot invoke "String.length()" because "this.input" is null
    at java.base/java.net.URI$Parser.parse(URI.java:3194)
    at java.base/java.net.URI.<init>(URI.java:649)
    ...
```

## Root Cause (hypothesis)

`CodeEvaluator` catches the `jdk.jshell.EvalException` thrown by JShell and re-wraps it in a plain `RuntimeException`. In doing so, it likely reconstructs only one level of cause rather than recursively walking `EvalException.getCause()` to rebuild the full chain.

The relevant location appears to be `CodeEvaluator.evalSingle()` (line 145).

## Impact

Without the root cause, debugging exceptions that originate inside third-party libraries is practically impossible from inside the notebook. The user is forced to reproduce the error outside of Jupyter to get a useful stack trace.

--- a/jjava-kernel/src/main/java/org/dflib/jjava/kernel/JavaKernel.java
+++ b/jjava-kernel/src/main/java/org/dflib/jjava/kernel/JavaKernel.java
@@ -127,6 +127,8 @@ public class JavaKernel extends BaseKernel {
             return formatEvaluationTimeoutException((EvaluationTimeoutException) e);
         } else if (e instanceof EvaluationInterruptedException) {
             return formatEvaluationInterruptedException((EvaluationInterruptedException) e);
+        } else if (e instanceof RuntimeException && e.getCause() instanceof EvalException) {
+            return formatEvalException((EvalException) e.getCause());
         } else {
             return new ArrayList<>(super.formatError(e));
         }
@@ -180,14 +182,28 @@ public class JavaKernel extends BaseKernel {
     private List<String> formatEvalException(EvalException e) {
         List<String> fmt = new ArrayList<>();
 
-
-        String evalExceptionClassName = EvalException.class.getName();
         String actualExceptionName = e.getExceptionClassName();
-        super.formatError(e).stream()
-                .map(line -> line.replace(evalExceptionClassName, actualExceptionName))
-                .forEach(fmt::add);
+        fmt.add(errorStyler.secondary(actualExceptionName + ": " + e.getMessage()));
+        for (StackTraceElement element : e.getStackTrace()) {
+            fmt.add(errorStyler.secondary("\tat " + element));
+        }
 
+        formatEvalExceptionCause((EvalException) e.getCause(), fmt);
         return fmt;
+    }
+
+    private void formatEvalExceptionCause(EvalException e, List<String> fmt) {
+        if (e == null) {
+            return;
+        }
+
+        String actualExceptionName = e.getExceptionClassName();
+        fmt.add(errorStyler.secondary("Caused by: " + actualExceptionName + ": " + e.getMessage()));
+        for (StackTraceElement element : e.getStackTrace()) {
+            fmt.add(errorStyler.secondary("\tat " + element));
+        }
+        
+        formatEvalExceptionCause((EvalException) e.getCause(), fmt);
     }
 
     private List<String> formatUnresolvedReferenceException(UnresolvedReferenceException e) {

--- a/jjava-kernel/src/main/java/org/dflib/jjava/kernel/execution/JJavaExecutionControl.java
+++ b/jjava-kernel/src/main/java/org/dflib/jjava/kernel/execution/JJavaExecutionControl.java
@@ -131,6 +131,18 @@ class JJavaExecutionControl extends DirectExecutionControl {
         return id;
     }
 
+    private RunException wrapInRunException(Throwable cause) {
+        if (cause instanceof SPIResolutionException) {
+            return new ResolutionException(((SPIResolutionException) cause).id(), cause.getStackTrace());
+        }
+
+        UserException ue = new UserException(String.valueOf(cause.getMessage()), cause.getClass().getName(), cause.getStackTrace());
+        if (cause.getCause() != null) {
+            ue.initCause(wrapInRunException(cause.getCause()));
+        }
+        return ue;
+    }
+
     private Object doInvoke(String id, Method doitMethod) throws Exception {
 
         Future<Object> task = isNestedCall()
@@ -160,10 +172,8 @@ class JJavaExecutionControl extends DirectExecutionControl {
             }
             if (cause == null) {
                 throw new UserException("null", "Unknown Invocation Exception", e.getStackTrace());
-            } else if (cause instanceof SPIResolutionException) {
-                throw new ResolutionException(((SPIResolutionException) cause).id(), cause.getStackTrace());
             } else {
-                throw new UserException(String.valueOf(cause.getMessage()), cause.getClass().getName(), cause.getStackTrace());
+                throw wrapInRunException(cause);
             }
         } catch (TimeoutException e) {
             String message = String.format("Execution timed out after configured timeout of %d %s.",

